### PR TITLE
GIF: Reimplement GIF FIFO to enable only when it is needed.

### PR DESCRIPTION
**Pull request by refractionpcsx2**
### Description of Changes
Reworks the GIF FIFO to always be enabled and only kick in when needed.  Savestate version bumped

### Rationale behind Changes
The old GIF FIFO was a bit janky/broken and it was quite slow when used, the new changes allow it to always be on without impacting performance.

### Suggested Testing Steps
Test mainly games which used to have the GIF FIFO hack entry in the GameDB, or games which use PATH3 masking, these will be impacted the most

### CODE REVIEW

**Comment by lightningterror** [comment]
Once changes are ready it would be nice to run clang on the files for consistent formatting.

File: pcsx2/FiFo.cpp [hunk 1]
```diff
@@ -125,10 +125,10 @@ void __fastcall WriteFIFO_VIF1(const mem128_t *value)
 void __fastcall WriteFIFO_GIF(const mem128_t *value)
 {
 	GUNIT_LOG("WriteFIFO_GIF()");
-	if (CHECK_GIFFIFOHACK) {
-		gif_fifo.write((u32*)value, 1);
-
-		gif_fifo.read(true);		
+	if ((!gifUnit.CanDoPath3() || gif_fifo.fifoSize > 0)) {
```
Comment (lightningterror) [inline pcsx2/FiFo.cpp:128]: This file didn't get formatted, oh well another time.

Comment (refractionpcsx2) [reply pcsx2/FiFo.cpp:128]: Happy now? :P

Comment (lightningterror) [reply pcsx2/FiFo.cpp:128]: Bellissimo.

**Comment by HeroponRikiBestest** [comment]
Breaks skipping the opening FMV in Air Ranger 2.

**Comment by refractionpcsx2** [comment]
Fixed in #5324

**Pull request merged**